### PR TITLE
Add some action items to the peer errors when node_modules is messed up

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Resolves: https://github.com/embroider-build/embroider/issues/1602

### CODE REVIEW

File: packages/compat/src/standalone-addon-build.ts [hunk 1]
```diff
@@ -29,6 +29,11 @@ export function convertLegacyAddons(compatApp: CompatApp) {
         `Some V1 ember addons are resolving as incorrect peer dependencies. This makes it impossible for us to safely convert them to v2 format.
 See https://github.com/embroider-build/embroider/blob/main/docs/peer-dependency-resolution-issues.md for an explanation of the problem and suggestions for fixing it.
 
+tl;dr: action items from the above documentation
+  - ensure your package-manager version is up to date
+  - delete all node_modules directories and re-install your dependencies
```
Comment (void-mAlex) [inline packages/compat/src/standalone-addon-build.ts:34]: would in most cases also need to remove your lock file

as a whole I don't feel confident this is the correct advice to give out as it lacks context of how a particular package manager behaves and which one of the many problems you could be encountering this may or may not fix;

doing a reroll of your dependencies could actually be harmful depending on your declared ranges in in package json and send people chasing down different non-errors

tl;dr: it's complicated and doing these steps without understanding why, likely not helpful

Comment (NullVoxPopuli) [reply packages/compat/src/standalone-addon-build.ts:34]: > would in most cases also need to remove your lock file

real projects can't seriously consider this option. It's perfectly fine in OSS tho

> [remove lock file] as a whole I don't feel confident this is the correct advice to give out

I agree :) 

> tl;dr: it's complicated and doing these steps without understanding why, likely not helpful

if the lockfile removal was included, sure, but the 3 steps listed "should be safe"

**Review by void-mAlex — APPROVED** [review]
